### DISCUSSION TITLE
What does the CoC mean at technical workshops, and how do we teach instructors to enforce it?

### DISCUSSION THREAD

**Discussion by JCSzamosi** [carpentries/trainers#147] [Instructional Content]
I'm wondering how people are talking about CoC violations and enforcement at their instructor trainings. 

Prepping for the training we're doing this week, I'm noticing that the guidelines for responding to CoC violations say very little about responding to incidents in the moment. The only in the moment action suggested is calling first responders/police, and then it moves straight to things like reporting to committee. I know the CoC guidelines are under active development, and I don't want to rush that process, but even as a temporary measure I'm pretty uncomfortable with "call the cops" being the only in-the-moment suggestion. I'm interested in knowing how people are teaching this section during instructor trainings, and also wondering if we can add a few more options into the guidelines as a stop-gap while they are developed, so we're not each making it up on the fly.

**Comment by marklcrowe** [comment]
This topic came up in our IT last week, although @JCSzamosi I'm not sure where you're seeing the recommendation about the police - I can't find it in the workshop material (https://carpentries.github.io/instructor-training/21-management/index.html) or the CoC page (https://docs.carpentries.org/topic_folders/policies/code-of-conduct.html). 

We reiterated part 2.3 of the CoC, especially "the organisers may warn the offender, ask them to leave the event or platform"; we felt it was very important that instructors recognise that they have the absolute right to ask someone to leave and they are under no obligation to deal with inappropriate behaviour further or make any other allowances. The Carpentries is not a service industry where you're often expected to put up with bad behaviour just because someone's a customer; we're a community with community standards that all members are expected to observe.

We also talked about having the responsibility of talking to the offender assigned to someone as part of pre-workshop planning; many of our instructors may not feed comfortable 'confronting' someone in this way, so it's important to identify who on the instructor/helper team is willing to take on that responsibility.

**Reply by JCSzamosi** [reply]
The part that particularly concerned me is that there is an exercise in the the workshop material that you linked which sends trainees to the incident response checklist ([here](https://docs.carpentries.org/topic_folders/policies/incident-response.html)), which does not anywhere mention asking people to leave a workshop or asking them to stop their behaviour, but only talks about physical safety and contacting EMS or law enforcement.

**Reply by JCSzamosi** [reply]
I see now the wording you mentioned in 2.3 of the CoC, but if a person is looking up how to respond to an incident (as I was), they are likely to look at the various incident response guidelines. I think it would be better if there was some of that language there as well.

**Comment by maneesha** [comment]
Tagging the @carpentries/code-of-conduct-committee and @fmichonneau  as the Core Team CoC liaison for input on this.

**Comment by yabellini** [comment]
I don't know if this directly answers the question, but it is related.  
I recommend reading this book on enforcing a Code of Conduct: https://frameshiftconsulting.com/code-of-conduct-book/.

Another thing we have done in MetaDocencia is to present different situations that can occur in the classroom so that we can reflect and discuss together to see how we could react and what to do. I learned in the RStudio certification and the Workshop for being a better ally by Open Life Science (https://openlifesci.org/posts/2020/10/05/ally-skills-training/) that people who have thought at least once in a situation like this are more likely to do something and do it effectively. Hence, it seems that community discussion and trainers' meetings could be excellent spaces for this kind of "training."

**Comment by yabellini** [comment]
I promised to share scenarios we used in training given in Spanish. The scenarios are for the DEI module, so not all of them have to do with CoC, but I hope they are helpful. In Spanish, sorry.

Scenario 1:
Eres jefe de trabajos prácticos en una clase de ingeniería de sistemas. El docente a cargo manifiesta su preocupación por el poco número de estudiantes mujeres en la carrera, asi que provee ayuda extra a las mujeres durante las prácticas cuando están trabajando en grupos pequeños y evita llamar a las mujeres a participar en clase para no exponerlas a todo el grupo.
¿Qué piensas de estas actitudes del docente? ¿Darán resultado? ¿Por qué sí o por qué no?

Scenario 2:
Sos ayudante de un nuevo profesor de termodinámica. El primer día de clase, el nuevo docente le dice a los estudiantes: “Este es un curso muy difícil. Van a tener que trabajar muchísimo más de lo que han trabajado antes para otros cursos y aún así un tercio de ustedes no va a aprobarlo”
¿Cómo te parece que los estudiantes pueden tomar este mensaje?
¿Qué pasos tomarías para hablar con el docente sobre el mensaje?

Scenario 3:
Estás dando un taller y les has pedido a tus estudiantes que trabajen en parejas. Después de unos minutos, el estudiante A se te acerca y te dice que su compañero, el estudiante B, ha estado haciendo bromas homofobicas. El estudiante B escucha esto e inmediatamente niega haberlo hecho. ¿Qué pasos tomarías ante esta situación?.  Asumamos que tu primer y segundo paso no resuelve la situación, piensa en un tercer y cuarto paso.

Scenario 4 (from Greg's slides): 
Un estudiante de tu clase tiene un déficit de atención diagnosticado médicamente. Para mantener la concentración, tiene que hablar consigo mismo mientras resuelve problemas. Otras/os estudiantes de la clase se te han acercado y te han dicho que les distrae. ¿Qué pasos tomarías y en qué orden?

Scenario 5:
Estas enseñando un taller durante el fin de semana en la facultad de ciencias naturales.  Una estudiante comenta que los dataset usados como ejemplos no son significativos, que le gustaría trabajar con otro tipo de datos. Cuando le preguntas que tipo de dataset le gustaría usar, otro estudiante contesta "seguro que alguno de esos datos que usan las feministas, siempre todo tiene que ver con el patriarcado con ella". ¿Qué tipo de intervención te parece que podes hacer como docente ante esta situación?

Scenario 6:
Durante un taller el docente pide a la clase que den ideas para solucionar un problema de un caso de estudio. Un estudiante migrante y de color hace una sugerencia, pero nadie la tiene en cuenta. Más tarde, otro estudiante varón blanco hace la misma sugerencia y el docente le reconoce el mérito. ¿Qué podrías hacer como ayudante? ¿Qué podrías hacer como estudiante? ¿Qué podrías hacer como docente para evitar que esta situación te ocurra?

Scenario 7:
Vas a iniciar un taller y antes de comenzar una de las personas que va a tomar el curso se acerca y te dice que quisiera que cuando te presentes aclares tus pronombres y le pidas al resto de la clase que haga lo mismo.  También que se sentiría mejor si puedas usar lenguaje inclusivo cuando enseñes. ¿Qué le contestarías? ¿Cómo crees que el resto de la clase reaccionará a este pedido? ¿Qué acciones tomarías si otros estudiantes se quejan de este pedido? 

Scenario 8: 
Estas explicando el codigo de conducta antes de iniciar el curso y uno de los estudiantes te interrumpe y te dice "en que clase de mundo viven ustedes que necesitan tener estos códigos de conducta para hacer estos talleres". ¿Qué tres respuestas le darías/acciones tomarías asumiendo que la acciones tomadas no dieron resultado?

**Reply by JCSzamosi** [reply]
Thanks Yani! Me and google translate took a stab at translations (since I was doing it for my own benefit anyway). I hope that's okay. More than happy to have these fixed by people who actually speak Spanish :)

Scenario 1:
```
Eres jefe de trabajos prácticos en una clase de ingeniería de sistemas. El docente a cargo manifiesta su preocupación por el poco número de estudiantes mujeres en la carrera, asi que provee ayuda extra a las mujeres durante las prácticas cuando están trabajando en grupos pequeños y evita llamar a las mujeres a participar en clase para no exponerlas a todo el grupo.
¿Qué piensas de estas actitudes del docente? ¿Darán resultado? ¿Por qué sí o por qué no?
```

You are the (head of labs? lab TA?) in a systems engineering class. The teacher in charge expresses his concern about the small number of female students in the course, and so provides extra help to the women during the (labs?) when they are working in small groups, and avoids calling on women to participate in class, so that he doesn't put them on the spot in front of the whole group. What do you think of these attitudes? Will they work? Why or why not?

Scenario 2
```
Sos ayudante de un nuevo profesor de termodinámica. El primer día de clase, el nuevo docente le dice a los estudiantes: “Este es un curso muy difícil. Van a tener que trabajar muchísimo más de lo que han trabajado antes para otros cursos y aún así un tercio de ustedes no va a aprobarlo”
¿Cómo te parece que los estudiantes pueden tomar este mensaje?
¿Qué pasos tomarías para hablar con el docente sobre el mensaje?
```

You are the TA for a new thermodynamics professor. On the first day of class, the new teacher says to the students: "This is a very difficult course. You will have to work much harder than you have had to work in previous courses, and even then a third of you will not pass." What do you think the students will take from this statement? What steps would you take to talk to the professor about their statement?

Scenario 3:
```
Estás dando un taller y les has pedido a tus estudiantes que trabajen en parejas. Después de unos minutos, el estudiante A se te acerca y te dice que su compañero, el estudiante B, ha estado haciendo bromas homofobicas. El estudiante B escucha esto e inmediatamente niega haberlo hecho. ¿Qué pasos tomarías ante esta situación?. Asumamos que tu primer y segundo paso no resuelve la situación, piensa en un tercer y cuarto paso.
```

You are giving a workshop and you have asked your students to work in pairs. After a few minues, student A approaches you and tells you that their partner, student B, has been making homophobic jokes. Student B hears this and immediately denies having done so. What steps would you take to address this situation? Assuming that your first and second steps don't resolve it, think of a third and fourth step.

Scenario 4 (from Greg's slides):
```
Un estudiante de tu clase tiene un déficit de atención diagnosticado médicamente. Para mantener la concentración, tiene que hablar consigo mismo mientras resuelve problemas. Otras/os estudiantes de la clase se te han acercado y te han dicho que les distrae. ¿Qué pasos tomarías y en qué orden?
```

A student in your class has a diagnosed attention deficit. In order to stay focused, they need to talk to themselves while they figure things out. Other students in the class have complained that this is distracting. What steps would you take, and in what order?

Scenario 5:
```
Estas enseñando un taller durante el fin de semana en la facultad de ciencias naturales. Una estudiante comenta que los dataset usados como ejemplos no son significativos, que le gustaría trabajar con otro tipo de datos. Cuando le preguntas que tipo de dataset le gustaría usar, otro estudiante contesta "seguro que alguno de esos datos que usan las feministas, siempre todo tiene que ver con el patriarcado con ella". ¿Qué tipo de intervención te parece que podes hacer como docente ante esta situación?
```

You are teaching a workshop over the weekend in the Faculty of Natural Sciences. A student comments that the example datasets are not (relevant? important?) and that she would like to work with a different kind of data. When you ask her what kind of data she would like to work with instead, another student comments "Probably some feminist data set. Everything is always about the patriarchy with her." What could you, as the teacher, do about this situation?"

Scenario 6:
```
Durante un taller el docente pide a la clase que den ideas para solucionar un problema de un caso de estudio. Un estudiante migrante y de color hace una sugerencia, pero nadie la tiene en cuenta. Más tarde, otro estudiante varón blanco hace la misma sugerencia y el docente le reconoce el mérito. ¿Qué podrías hacer como ayudante? ¿Qué podrías hacer como estudiante? ¿Qué podrías hacer como docente para evitar que esta situación te ocurra?
```

During a workshop, the teacher asks the class for ideas to solve a problem in a case study. An immigrant student of colour has a suggestion, but no one takes it seriously. Later on, a white, male student makes the same suggestion and the teacher acknowledges its merits. What can you do if you're a TA? What can you do if you're a student? What could you do as the teacher to avoid being in this situation in the first place?

Scenario 7:
```
Vas a iniciar un taller y antes de comenzar una de las personas que va a tomar el curso se acerca y te dice que quisiera que cuando te presentes aclares tus pronombres y le pidas al resto de la clase que haga lo mismo. También que se sentiría mejor si puedas usar lenguaje inclusivo cuando enseñes. ¿Qué le contestarías? ¿Cómo crees que el resto de la clase reaccionará a este pedido? ¿Qué acciones tomarías si otros estudiantes se quejan de este pedido?
```

You are about to start a workshop. Before you begin, one of the learners approaches you and tells you that they would like you to give your pronouns when you introduce yourself and to ask the rest of the class to do the same. They also say that they would feel better if you could use inclusive language while you're teaching. How would you respond? How do you think the rest of the class would react to this question? What actions could you take if other students object to the question?

Scenario 8:
```
Estas explicando el codigo de conducta antes de iniciar el curso y uno de los estudiantes te interrumpe y te dice "en que clase de mundo viven ustedes que necesitan tener estos códigos de conducta para hacer estos talleres". ¿Qué tres respuestas le darías/acciones tomarías asumiendo que la acciones tomadas no dieron resultado?
```

You are explaining the code of conduct before starting a course, and one of the students interrupts you and says "what kind of world do you live in that you need this code of conduct in order to hold these workshops?" What three responses could you give/actions could you take assuming that the actions taken did not resolve the situation?

**Reply by yabellini** [reply]
They are good translation. Thanks!!

 I think we can use TA for _jefe de trabajos practicos_ because I don't know the equivalent of this teaching position at universities outside Argentina.

About the dataset relevant can be a good word.

**Comment by karinlag** [comment]
These are very good! Thanks so much for both versions!

**Comment by murraycadzow** [comment]
Something that was raised in our recent workshop, and is becoming more relevant with many workshops being taught online is that it would be good to have more guidance on how to enforce the CoC online for in-the-moment responses
